### ISSUE TITLE
Search does not work when site BaseURL is a subdirectory

### ISSUE THREAD

**Issue by jborlik** [tract-docs/tract#8]
I need to have the Hugo-generated site be located in a subdirectory, e.g. "https://example.com/something1/something2/myhugosite".  As far as I can tell, the way to do that is to define baseURL in my site's config.toml.  (Maybe that's not right?  I'm new to Hugo.)

However, if I define baseURL, search does not seem to return anything.  I notice from Chrome Developer Tools that the js is looking for "/index.json" (and not able to find it) rather than "/something1/something2/myhugosite/index.json".  

If I change layouts/partials/search.html line 2:

```
{{ $indexUrl := cond $isI18n (printf "%s/%s/index.json" site.BaseURL .Lang) (printf "%s/index.json" site.BaseURL)  }}
```

and line 85:

```
const item = result( '{{ site.BaseURL }}' + res.url, res.title, res.section);
```

This seems to get search to work in my scenario.  But I don't know if this the breaks something else, or if I'm just configuring things wrong.

**Comment by lucperkins** [comment]
@jborlik Do you by chance have a repo I can look at?

**Comment by jborlik** [comment]
Sorry, @lucperkins, that I haven't responded sooner.  I tried to replicate it by starting a new site, following the Tract "getting started" instructions, and then setting `baseurl="http://example.com/subdir/mysite"` in the new site's config.toml... and everything seemed to work as expected.  So I don't know.  

I notice when I set `canonifyURLs = true`, I get Chrome console errors that show index.json was not found.  But now I don't know why that was set in the my original site!
